### ISSUE TITLE
UI: Allow for specifying a specific git commit for a test

### ISSUE THREAD

**Issue by fivenp** [kubeshop/testkube#3157]
**Describe the enhancement you'd like to see**
Users can currently define either a git branch or a git commit in the source for their tests. 
The dashboard lacks this functionality (#3105 )

We want to add a switchbox to the test EDIT screen (only the edit screen!) where the user can select whether his git source (File AND Dir) is referring to a specific branch or a commit.

What we have today: 
<img width="172" alt="image" src="https://user-images.githubusercontent.com/132332/217497898-d0963704-3234-4938-a756-33d188572644.png">

What we want to have: 

![image](https://user-images.githubusercontent.com/132332/217497937-70420205-7235-4d46-994f-f7f067a0054d.png)

<img width="314" alt="image" src="https://user-images.githubusercontent.com/132332/217498398-4ce3acbe-252a-41da-a4d3-c4e68b9e63bf.png">

**Comment by TheBrunoLopes** [comment]
Good catch. anything else we might be missing on the Test Creating screen ?

**Comment by fivenp** [comment]
> Good catch. anything else we might be missing on the Test Creating screen ?

This is NOT going to be part of the test creation! (because it will blow the already big form even more!) 
The use case for git commit is so low – that we only allow you to switch in the test edit

**Comment by TheBrunoLopes** [comment]
@fivenp review

**Comment by fivenp** [comment]
Reviewed and good to go
